pkg/app/piped/controller/controller.go

```diff
@@ -219,6 +224,16 @@ func (c *controller) Run(ctx context.Context) error {
 		close(lpStoppedCh)
 	}()
 
+	// Start workspace cleaner.
+	// This will remove the workspace directory of the completed planner/scheduler.
+	go func() {
+		for ws := range c.workingDirRemovalCh {
```

Comment:
[Ask] It looks like there is nothing to close `c.workingDirRemovalCh`. Does this lead to the goroutine leak?

Reply:
I'm not sure if we would call this pattern a goroutine leak.
This goroutine's lifetime is the same as that of the piped.
As there is only one controller for each piped, there is also only one instance of this goroutine.
I believe this code is acceptable because the number of goroutines doesn't increase.

Reply:
@Warashi 

> This goroutine's lifetime is the same as that of the piped.
As there is only one controller for each piped, there is also only one instance of this goroutine.

I see. I got your point 👍 
But I think it would be better to make the lifetime of the channel explicit.

It would be better for the person calling controller.Run to not be aware of that.

WDYT?

Reply:
> It would be better for the person calling controller.Run to not be aware of that.

I got your point. I'll try to make the lifetime explicit.

Reply:
Fixed in this commit
https://github.com/pipe-cd/pipecd/pull/5217/commits/efa4e6d60cca68d7fa8b35896f5a9ad0290ce9e9